airflow_dbt/hooks/google.py

```diff
@@ -0,0 +1,146 @@
+        if not staging_blob.endswith('.tar.gz'):
+            raise AirflowException(
+                f'The provided blob "{staging_blob}" to a compressed file does not ' +
+                'have the right extension ".tar.gz'
+            )
+        self.gcs_staging_bucket = staging_bucket
+        self.gcs_staging_blob = staging_blob
+
+        self.dbt_version = dbt_version
+        self.cloud_build_hook = CloudBuildHook(gcp_conn_id=gcp_conn_id)
+        self.gcp_conn_id = gcp_conn_id
+        self.project_id = self.cloud_build_hook.project_id if project_id is None else project_id
+        self.service_account = service_account
+
+        super().__init__(env=env)
+
+    def get_conn(self) -> Any:
+        """Returns the cloud build connection, which is a gcp connection"""
+        return self.cloud_build_hook.get_conn()
+
+    def run_dbt(self, dbt_cmd: List[str]):
+        """
+         Run the dbt command. In version 5 of the providers
+
+         :param dbt_cmd: The dbt whole command to run
+         :type dbt_cmd: List[str]
+         """
+        # See: https://cloud.google.com/cloud-build/docs/api/reference/rest/v1/projects.builds
+        cloud_build_config = {
+            'steps': [{
+                # use the official dbt docker image from dockerhub
+                'name': f'fishtownanalytics/dbt:{self.dbt_version}',
+                'args': dbt_cmd,
+                'env': [f'{k}={v}' for k, v in self.env.items()]
+            }],
+            'source': {
+                'storageSource': {
+                    "bucket": self.gcs_staging_bucket,
+                    "object": self.gcs_staging_blob,
+                }
+            },
+            'options': {
+                # default is legacy and its behaviour is subject to change
+                'logging': 'GCS_ONLY',
+            },
+            # mandatory if using a service_account, it also is relevant as
+            # transactional data
+            'logsBucket': self.gcs_staging_bucket,
+        }
+
+        if self.service_account is not None:
+            sa_path = f'projects/{self.project_id}/serviceAccounts/' \
+                      f'{self.service_account}'
+            cloud_build_config['serviceAccount'] = sa_path
+
+        cloud_build_config_str = json.dumps(cloud_build_config, indent=2)
+        logging.info(
+            f'Running the following cloud build'
+            f' config:\n{cloud_build_config_str}'
+        )
+
+        build_results = self.cloud_build_hook.create_build(
+            body=cloud_build_config,
+            project_id=self.project_id,
+        )
+
+        # print logs from GCS
+        build_logs_blob = f'log-{build_results["id"]}.txt'
+        with GCSHook().provide_file(
+            bucket_name=self.gcs_staging_bucket,
+            object_name=build_logs_blob
+        ) as log_file_handle:
+            for line in log_file_handle:
+                clean_line = line.decode('utf-8').strip()
```

Comment:
This will be integrated in the airflow-google-provider v6, not yet supported by .... Google Cloud themselves :D  

It will be free then. Provided by the library. It's not the prettiest but that's what we can do for now :/